dotCMS/src/main/java/com/dotmarketing/servlets/ImportExportXMLServlet.java

```diff
@@ -182,30 +179,24 @@ private void doUpload(HttpServletRequest request, HttpServletResponse response)
 		String tempdir = FileUtil.getRealPath(backupTempFilePath);
 
 		MultipartRequest mpr;
-		try {
-			mpr = new MultipartRequest(request, tempdir, 1000000000);
-			File importFile = mpr.getFile("fileUpload");
-
-			/*
-			 * Unzip zipped backups
-			 */
-			if (importFile != null && importFile.getName().toLowerCase().endsWith(".zip")) {
-
-				InputStream in = new BufferedInputStream(new FileInputStream(importFile));
-				ZipInputStream zin = new ZipInputStream(in);
-				ZipEntry e;
-
-				while ((e = zin.getNextEntry()) != null) {
-					unzip(zin, e.getName());
-				}
-				zin.close();
-				importFile.delete();
-			}
-
-		} catch (IOException e) {
-			// TODO Auto-generated catch block
-			Logger.error(this,e.getMessage(),e);
-		}
+        mpr = new MultipartRequest(request, tempdir, 1000000000);
+        File importFile = mpr.getFile("fileUpload");
+
+        /*
+         * Unzip zipped backups
+         */
+        if (importFile != null && importFile.getName().toLowerCase().endsWith(".zip")) {
+
+            try(final InputStream in = new BufferedInputStream(Files.newInputStream(importFile.toPath()));
+                    ZipInputStream zin = new ZipInputStream(in)){
+                ZipEntry e;
+
+                while ((e = zin.getNextEntry()) != null) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a "try/catch" block for "getNextEntry". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1989)